### PR TITLE
Add document for CTE

### PR DESCRIPTION
**Pull request by Liuxiaozhen12**
### What is changed, added or deleted? (Required)
Close https://github.com/pingcap/docs-cn/issues/6216

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/6220 https://github.com/pingcap/docs-cn/pull/6265
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by morgo — APPROVED** [review]

File: sql-statements/sql-statement-with.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+| 5 |
++---+
+5 rows in set (0.00 sec)
+```
+
+## MySQL compatibility
+
+* In strict mode, when the data length recursively calculated exceeds the data length of the seed part, TiDB will return a warning while MySQL will return an error. In non-strict mode, the behavior of TiDB is consistent with that of MySQL.
+* The data type for recursive CTE is determined by the seed part. The data type of the seed part is not completely consistent with MySQL in some cases (such as functions).
+* In the case of multiple UNION / UNION ALL, MySQL does not allow UNION to be followed by UNION ALL, TiDB does.
```
Comment (TomShawn) [inline sql-statements/sql-statement-with.md:87]:
```suggestion
* In the case of multiple `UNION` / `UNION ALL` operators, MySQL does not allow `UNION` to be followed by `UNION ALL`, but TiDB does.
```

**Review by TomShawn — APPROVED** [review]
Rest LGTM

File: sql-statements/sql-statement-with.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
+---
+title: WITH | TiDB SQL Statement Reference
+summary: An overview of the usage of WITH (Common Table Expression) for the TiDB database.
+---
+
+# WITH
+
+A Common Table Expression (CTE) is a temporary result set that can be referred possibly multiple times within the scope of a SQL statement to promote its readability and execution efficiency. You can apply the `WITH` statement to use Common Table Expressions.
```
[the remark this answers is not in the document]
Comment (morgo) [reply sql-statements/sql-statement-with.md]: I think you can even simplify it to say "can be refered to within a SQL statement" - the multiple times is a more advanced detail that can be omitted for brevity.

File: sql-statements/sql-statement-with.md [hunk 3]
```diff
@@ -82,9 +82,9 @@ WITH RECURSIVE cte(a) AS (SELECT 1 UNION SELECT a+1 FROM cte WHERE a < 5) SELECT
 
 ## MySQL compatibility
 
-* In strict mode, when the data length recursively calculated exceeds the data length of the seed part, TiDB will return a warning while MySQL will return an error. In non-strict mode, the behavior of TiDB is consistent with that of MySQL.
+* In the strict mode, when the data length recursively calculated exceeds the data length of the seed part, TiDB returns a warning while MySQL returns an error. In the non-strict mode, the behavior of TiDB is consistent with that of MySQL.
```
Comment (morgo) [inline sql-statements/sql-statement-with.md:85]: @TomShawn MySQL refers to it as "strict mode" not "the strict mode". See: https://dev.mysql.com/doc/refman/8.0/en/sql-mode.html#sql-mode-strict for example.

File: sql-statements/sql-statement-with.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+| 3 |
+| 4 |
+| 5 |
++---+
+5 rows in set (0.00 sec)
+```
+
+## MySQL compatibility
+
+* In the strict mode, when the data length recursively calculated exceeds the data length of the seed part, TiDB returns a warning while MySQL returns an error. In the non-strict mode, the behavior of TiDB is consistent with that of MySQL.
```
Comment (TomShawn) [inline sql-statements/sql-statement-with.md:85]:
```suggestion
* In strict mode, when the data length recursively calculated exceeds the data length of the seed part, TiDB returns a warning while MySQL returns an error. In non-strict mode, the behavior of TiDB is consistent with that of MySQL.
```

File: sql-statements/sql-statement-with.md (hunk 3 above)
Comment (TomShawn) [reply sql-statements/sql-statement-with.md]: Removed in https://github.com/pingcap/docs/pull/5770/commits/9a1cf22b268b3865bf427c301d491878531fbab5. Thanks for letting me know.

**Pull request merged**